addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/handler/AccountHandler.java

```diff
@@ -0,0 +1,635 @@
+import org.openhab.binding.amazonechocontrol.internal.ConnectionException;
+import org.openhab.binding.amazonechocontrol.internal.LoginServlet;
+import org.openhab.binding.amazonechocontrol.internal.StateStorage;
+import org.openhab.binding.amazonechocontrol.internal.discovery.AmazonEchoDiscovery;
+import org.openhab.binding.amazonechocontrol.internal.discovery.IAmazonEchoDiscovery;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonBluetoothStates;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonBluetoothStates.BluetoothState;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonDevices.Device;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonFeed;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonSmartHomeDevice;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Handles the connection to the amazon server.
+ *
+ * @author Michael Geramb - Initial Contribution
+ */
+public class AccountHandler extends BaseBridgeHandler implements IAmazonEchoDiscovery {
+
+    private final Logger logger = LoggerFactory.getLogger(AccountHandler.class);
+    private StateStorage stateStorage;
+    private AccountConfiguration config;
+    private Connection connection;
+    private List<EchoHandler> echoHandlers = new ArrayList<>();
+    private List<SmartHomeBaseHandler> smartHomeHandlers = new ArrayList<>();
+    private List<FlashBriefingProfileHandler> flashBriefingProfileHandlers = new ArrayList<>();
+    private Object synchronizeConnection = new Object();
+    private Map<String, Device> jsonSerialNumberDeviceMapping = new HashMap<>();
+    private ScheduledFuture<?> refreshJob;
+    private ScheduledFuture<?> refreshLogin;
+    private boolean updateSmartHomeDeviceList;
+    private boolean discoverFlashProfiles;
+    private boolean smartHodeDeviceListEnabled;
+    private String currentFlashBriefingJson = "";
+    private HttpService httpService;
+    private LoginServlet loginServlet;
+
+    public AccountHandler(@NonNull Bridge bridge, @NonNull HttpService httpService) {
+        super(bridge);
+        this.httpService = httpService;
+        stateStorage = new StateStorage(bridge);
+        AmazonEchoDiscovery.setHandlerExist();
+
+    }
+
+    @Override
+    public void initialize() {
+        start();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Command '{}' received for channel '{}'", command, channelUID);
+
+        if (command instanceof RefreshType) {
+            refreshData();
+        }
+    }
+
+    public Device[] getLastKnownDevices() {
+        Map<String, Device> temp = jsonSerialNumberDeviceMapping;
+        if (temp == null) {
+            return new Device[0];
+        }
+        Device[] devices = new Device[temp.size()];
+        temp.values().toArray(devices);
```

Comment:
Why don't you return the list as part of your API its way more standard then working with arrays?

Reply:
I'am not so experienced in java. Please give me an example!